common/beerocks/bwl/nl80211/sta_wlan_hal_nl80211.cpp

```diff
@@ -59,7 +203,21 @@ bool sta_wlan_hal_nl80211::unassoc_rssi_measurement(const std::string &mac, int
     return true;
 }
 
-bool sta_wlan_hal_nl80211::is_connected() { return true; }
+bool sta_wlan_hal_nl80211::is_connected()
+{
+    ConnectionStatus connection_status;
+    int retries = 10;
+    while (!read_status(connection_status)) {
+        --retries;
+        LOG(ERROR) << "is_connected - read_status failed, retry:" << 10 - retries;
+        if (retries == 0) {
+            break;
+        }
+        usleep(1000);
+    }
```

Comment:
```suggestion
    constexpr int MAX_RETRIES = 10;
    int retry = 0;
    while ((!read_status(connection_status)) && (retry++ < MAX_RETRIES)) {
        LOG(ERROR) << "Failed to read connection status, retry: " << retries;
        usleep(1000);
    }
```
1. since you don't have special handling (other than a `break`) when the max number of retries is reached, why not have it in the condition itself
2. the log print more straightforward this way

you can move the `retry++` inside the loop if you find it clearer.